Published-source: https://deb.freexian.com/extended-lts/pool/main/p/php7.0/php7.0_7.0.33-0+deb9u26.debian.tar.xz
Published-patch: 0102-Fix-bug-78599-env_path_info-underflow-can-lead-to-RC.patch
Scope: selected PHP runtime components and their tests; packaging and unrelated extensions omitted.

From: Jakub Zelenka <bukka@php.net>
Date: Sat, 12 Oct 2019 15:56:16 +0100
Subject: Fix bug #78599 (env_path_info underflow can lead to RCE)
 (CVE-2019-11043)

cheery-picked from ab061f95ca966731b1c84cf5b7b20155c0a1c06a
without the test as tester not available
---
 sapi/fpm/fpm/fpm_main.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/sapi/fpm/fpm/fpm_main.c
+++ b/sapi/fpm/fpm/fpm_main.c
@@ -1208,8 +1208,8 @@ static void init_request_info(void)
 								path_info = script_path_translated + ptlen;
 								tflag = (slen != 0 && (!orig_path_info || strcmp(orig_path_info, path_info) != 0));
 							} else {
-								path_info = env_path_info ? env_path_info + pilen - slen : NULL;
-								tflag = (orig_path_info != path_info);
+								path_info = (env_path_info && pilen > slen) ? env_path_info + pilen - slen : NULL;
+								tflag = path_info && (orig_path_info != path_info);
 							}
 
 							if (tflag) {
